dips/dip-158.md

```diff
@@ -111,35 +113,50 @@ Each scenario is expressed using a sequence of Sender / Receiver functionality.
 |-|-|-|-|
 |1|GetPaymentInfo|Sender|The wallet requests payment details using the reference id provided. The receiver will respond with the payment details and merchant data. The wallet will display the payment details so the customer can approve or reject|
 |2|InitChargePayment|Sender|Following customer approval the wallet initializes the payment process and sends the payer data|
-|3|ReadyForSettlementNotification|Sender|The wallet indicates the payment is ready for settlement. The wallet is expected to submit a transaction on-chain immediately before or after this step|
+|3|Put transaction on-chain|Sender|Following a successful response from the receiver (to the InitChargePayment in the previous step), the wallet must put a corresponding transaction on-chain and include the Reference ID as part of the metadata. See [appendix B](#appendix-b---include-reference-id-in-the-on-chain-transaction). If the payment is over the travel rule threshold, the sender must verify the recipient provided its signature as part of the InitChargePayment response|
+
+**Note** that following the completion of the Charge sequence, the receiver should start looking for the transaction put by the sender on the blockchain 
 
 #### Charge Sequence Diagram
-![Charge Sequence Diagram](https://static.swimlanes.io/13ec145db305a3c02323bb3288db52a3.png)
+TBD
```

Comment:
Let's wait until we have a sequence diagram before accepting/merging.

Reply:
I update the same diagram in swimlanes. 
You can always check the URL below for the "latest and greatest" of the Charge payment (QR scan) flow
https://swimlanes.io/u/CVMdULSTu

Let's agree on the finalized version and then we can save as image, upload to GitHub and embed in the DIP